vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractRow.java

```diff
@@ -413,8 +413,8 @@ private CELL moveColumnLayerAndJoinCells(Collection<CELL> cellsToJoin,
                 ColumnGroup group = ColumnGroupHelpers.wrapInColumnGroup(grid,
                         col);
                 AbstractColumn<?> oldGroup = leftColumns.next();
-                group.setHeaderRenderer(oldGroup.getHeaderRenderer());
-                group.setFooterRenderer(oldGroup.getFooterRenderer());
+                oldGroup.moveHeaderContent(group);
+                oldGroup.moveFooterContent(group);
```

Comment:
As the new logic for generating the aria-label depends on the header text / component being present, just moving the renderer is not enough. The new methods will assign either the text or the component to the other column / group, which creates a new renderer instead.